spec/example_app/app/dashboards/customer_dashboard.rb

```diff
@@ -8,6 +8,10 @@ class CustomerDashboard < Administrate::BaseDashboard
     email_subscriber: Field::Boolean,
     lifetime_value: Field::Number.with_options(prefix: "$", decimals: 2),
     name: Field::String,
+    nickname: Field::String.with_options(
+      getter: ->(field) { "Mr. #{field.resource.name}man" if field.resource.name.present? },
+      searchable: false
+    ),
```

Comment:
Thinking of an example that doesn't feel too artificial: how about a `full_address` field for `Order`? Something like this:

```ruby
    full_address: Field::String.with_options(
      getter: ->(field) {
        r = field.resource
        [
          r.address_line_one,
          r.address_line_two,
          r.address_city,
          r.address_state,
          r.address_zip
        ].compact.join("\n")
      },
      searchable: false
    ),
```

Then its `SHOW_PAGE_ATTRIBUTES` can be something like this:

```ruby
  SHOW_PAGE_ATTRIBUTES = FORM_ATTRIBUTES
    .except("address")
    .merge(
      "" => %i[customer full_address created_at updated_at],
      "details" => %i[line_items total_price shipped_at payments]
    )
    .freeze
```

Reply:
Thanks. I added.

Reply:
Looks like `searchable: false` is excess. If getter has been provided it makes field as unsearchable.

Reply:
Have you thought about using decorators(like draper) for virtual fields?

Reply:
@Nitr @pablobm 
Thank you for the advice. I have made the changes. https://github.com/thoughtbot/administrate/pull/2658/commits/cab60966f9eb8edca26c5bfd18b75962c59691eb
I'm not sure if we should always set `searchable` to `false` when a `getter` is present, but since I can't think of any use cases at the moment, this seems appropriate.